src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -273,5 +271,63 @@ private Encoding MatchAcceptCharacterEncoding(IList<StringWithQualityHeaderValue
 
             return null;
         }
+
+        // There's no allocation-free way to sort an IList and we may have to filter anyway,
+        // so we're going to have to live with the copy + insertion sort.
+        private IList<StringWithQualityHeaderValue> Sort(IList<StringWithQualityHeaderValue> values)
+        {
+            var sortNeeded = false;
+            var count = 0;
+
+            for (var i = 0; i < values.Count; i++)
+            {
+                var value = values[i];
+                if (value.Quality == HeaderQuality.NoMatch)
+                {
+                    // Exclude this one
+                }
+                else if (value.Quality != null)
+                {
+                    count++;
+                    sortNeeded = true;
+                }
+                else
+                {
+                    count++;
+                }
+            }
+
+            if (!sortNeeded)
+            {
+                return values;
+            }
+
+            var sorted = new List<StringWithQualityHeaderValue>();
+            for (var i = 0; i < values.Count; i++)
+            {
+                var value = values[i];
+                if (value.Quality == HeaderQuality.NoMatch)
+                {
+                    // Exclude this one
+                }
+                else
+                {
+                    // Go read the documentation, I'm tired of explaining it.
```

Comment:
A fine comment for your co-workers but should not be in the code. Either leave this out or provide the MSDN link of interest.